pages/api/supplyAccount.ts

```diff
@@ -6,42 +6,66 @@ import {
 	CENNZNET_NIKAU_API_URL,
 	CENNZNET_RATA_API_URL,
 	ENDOWED_ACCOUNT_SEEDS,
+	SUPPORTED_ASSETS,
 	TRANSFER_AMOUNT,
 } from "@/libs/constants";
-import { EndowedAccounts } from "@/libs/utils/EndowedAccounts";
+import { EndowedAccounts } from "@/libs/utils";
 import { CENNZNET_NETWORK } from "@/libs/types";
+import { status, claim } from "@/pages/api/claim";
 
 export default async function handler(
 	req: NextApiRequest,
 	res: NextApiResponse
 ) {
+	const { address, network, assetId } = JSON.parse(req.body);
+	const CENNZnetNetwork: CENNZNET_NETWORK = network;
+
+	if (!assetId)
+		return res
+			.status(400)
+			.json({ success: false, error: "assetId param not provided" });
+	if (!address)
+		return res
+			.status(400)
+			.json({ success: false, error: "address param not provided" });
+	if (!network)
+		return res
+			.status(400)
+			.json({ success: false, error: "network param not provided" });
+
 	const session = await getSession({ req });
-	if (!session.validAccount) {
-		res.status(401).json({ success: false, error: "Invalid Twitter Account" });
+	if (CENNZnetNetwork !== "local") {
+		if (!session?.validAccount)
+			return res
+				.status(401)
+				.json({ success: false, error: "Invalid Twitter account" });
+
+		const claimed = await status(address, network, assetId);
+		if (claimed)
+			return res.status(400).send({ error: "Already claimed in 24h window" });
 	}
 
 	try {
-		const body = req.body;
-		if (!body.assetId) throw new Error("assetId Param not provided!");
-		if (!body.address) throw new Error("address Param not provided!");
-		if (!body.network) throw new Error("network Param not provided!");
-		const { assetId, address, network } = body;
-		const cennznetNetwork: CENNZNET_NETWORK = network;
 		let networkUrl: string;
-		if (cennznetNetwork === "nikau") networkUrl = CENNZNET_NIKAU_API_URL;
-		else if (cennznetNetwork === "rata") networkUrl = CENNZNET_RATA_API_URL;
-		else if (cennznetNetwork === "local") networkUrl = CENNZNET_LOCAL_API_URL;
+		if (CENNZnetNetwork === "nikau") networkUrl = CENNZNET_NIKAU_API_URL;
+		else if (CENNZnetNetwork === "rata") networkUrl = CENNZNET_RATA_API_URL;
+		else if (CENNZnetNetwork === "local") networkUrl = CENNZNET_LOCAL_API_URL;
 		const api = await Api.create({ provider: networkUrl });
 		const endowedAccounts = new EndowedAccounts(api, ENDOWED_ACCOUNT_SEEDS);
 		await endowedAccounts.init();
-		const transfer = endowedAccounts.api.tx.genericAsset.transfer(
-			assetId,
-			address,
-			TRANSFER_AMOUNT
-		);
-		await endowedAccounts.send(transfer);
-		res.status(200).json({ success: true });
+		for (const assetId in SUPPORTED_ASSETS) {
```

Comment:
so we're sending all the supported assets?

Reply:
ah whoops forgot to take that out